api/src/main/java/io/opentelemetry/common/Attributes.java

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.common;
+
+import static java.util.Collections.emptySet;
+import static java.util.Collections.singleton;
+
+import com.google.auto.value.AutoValue;
+import com.google.auto.value.extension.memoized.Memoized;
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/** javadoc me. */
+@Immutable
+public abstract class Attributes implements Iterable<Entry<String, AttributeValue>> {
+  private static final Attributes EMPTY = new EmptyAttributes();
+
+  /** javadoc me. */
+  public abstract Set<String> keys();
```

Comment:
Is this useful?

Reply:
I can imagine situations where this is useful, but if we don't think we need it in the SDK (for metrics, maybe?), then I'm happy to remove any and all methods that aren't absolutely necessary for now.

Reply:
removed